src/ILLink.RoslynAnalyzer/ISymbolExtensions.cs

```diff
@@ -22,6 +22,46 @@ internal static bool HasAttribute (this ISymbol symbol, string attributeName)
 			return false;
 		}
 
+		internal static bool TryGetDynamicallyAccessedMemberTypes (this ISymbol symbol, out DynamicallyAccessedMemberTypes? dynamicallyAccessedMemberTypes)
+		{
+			dynamicallyAccessedMemberTypes = null;
+			if (!symbol.HasAttribute (DynamicallyAccessedMembersAnalyzer.DynamicallyAccessedMembersAttribute))
+				return false;
+
+			var damAttributeName = DynamicallyAccessedMembersAnalyzer.DynamicallyAccessedMembersAttribute;
+			AttributeData? dynamicallyAccessedMembers = null;
+			foreach (var _attribute in symbol.GetAttributes ())
+				if (_attribute.AttributeClass is var attrClass && attrClass != null &&
+					attrClass.HasName (damAttributeName)) {
+					dynamicallyAccessedMembers = _attribute;
+					break;
+				}
+
+			dynamicallyAccessedMemberTypes = (DynamicallyAccessedMemberTypes) dynamicallyAccessedMembers?.ConstructorArguments[0].Value!;
```

Comment:
This cast looks wrong. If this can be null it should be `(DynamicallyAccessMemberTypes?)`